backend/substrapp/tasks/tasks.py

```diff
@@ -625,6 +626,31 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
             client.volumes.create(name=volume_id)
         model_volume[volume_id] = {'bind': '/sandbox/local', 'mode': 'rw'}
 
+    if compute_plan_id is not None and settings.TASK['CHAINKEYS_ENABLED']:
+        chainkeys_directory = get_chainkeys_directory(compute_plan_id)
+
+        if not os.path.exists(chainkeys_directory):
+            secret_namespace = os.getenv('K8S_SECRET_NAMESPACE', 'default')
+            label_selector = f"compute_plan={subtuple.get('tag')}"
+
+            k8s_client = get_k8s_client()
+            try:
+                secrets = k8s_client.list_namespaced_secret(secret_namespace, label_selector=label_selector)
+            except ApiException as e:
+                logging.error(f'failed to fetch namespaced secrets {secret_namespace} with selector {label_selector}')
+                raise e
+
+            os.mkdirs(chainkeys_directory)
+            with open(path.join(chainkeys_directory, 'chainkeys.json'), 'w') as f:
+                json.dump(f, secrets)
```

Comment:
`secrets` will be a `V1SecretList` of objects of this kind:
```python
[
{           'api_version': None,
            'data': {'key': 'q1mjlIXJIEt6ADKrrHglXITCavH6AWq3sv83+XknsMs='},
            'kind': None,
            'metadata': {'annotations': None,
                         'cluster_name': None,
                         'creation_timestamp': datetime.datetime(2020, 1, 13, 10, 23, 4, tzinfo=tzlocal()),
                         'deletion_grace_period_seconds': None,
                         'deletion_timestamp': None,
                         'finalizers': None,
                         'generate_name': None,
                         'generation': None,
                         'initializers': None,
                         'labels': {'compute_plan': 'compute-plan-1'},
                         'managed_fields': None,
                         'name': 'compute-plan-1-a-next-ck',
                         'namespace': 'participant',
                         'owner_references': None,
                         'resource_version': '3622243',
                         'self_link': '/api/v1/namespaces/participant/secrets/compute-plan-1-a-next-ck',
                         'uid': '87bf9600-5529-4a3e-92b2-3a1a68cd256d'},
            'string_data': None,
            'type': 'Opaque'},
]
```
Dumping it to json will fail:
```
TypeError: Object of type 'V1SecretList' is not JSON serializable
```
`V1SecretList` are not iterable.
You need to first transform it:
```python
secrets = secrets.to_dict()['items']
```

Furthermore, I don't think this is really what we want here...
Secrets are base64 encoded, we need to base64 decode with `b64decode`

Updated code would be:
```python
secrets  = [{s['metadata']['name']: b64decode(s['data']['key'])} for s in secrets.to_dict()['items']]
```
You will then get a list of secrets with their name and their bytes values.
Each secret value is the representation of a very big integer to bytes.
You can load this value thanks to `int.from_bytes` method, this way:
```python
int.from_bytes(secret, 'big')
```

Remember json cannot dump bytes values, so you need to bring them back to int values:
Final code is:
```python
secrets = [{s['metadata']['name']: int.from_bytes(b64decode(s['data']['key']), 'big')} for s in secrets.to_dict()['items']]
json.dump(f, secrets)
```

Last thing to keep in mind, these secrets values do not incorporate their sign for the masking part, you need to compute them in the substra-backend or algo (or secure aggregation library) part.